File: 189-rotate-array/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        if (k % nums.length == 0)
            return;
        int i,j;
        int res = k % nums.length;
        int round=nums.length-res;
        int arr[]=new int[nums.length];
        for (i = round, j=0; i < nums.length; i++, j++){
            arr[j] = nums[i];
        }
        System.out.println(j);
        for (int m = 0; m < round; m++,j++)
            arr[j] = nums[m];

        for (int l = 0; l < nums.length; l++)
            nums[l] = arr[l];
    }
}

File: 189-rotate-array/rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int res = k % nums.size();
        int j = nums.size()-1;
        int temp;
        for (int i = 0; i < j; i++, j--){
            temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
        }
        int split = nums.size() - res;

        for (int a = 0, b = res-1; a < b; a++, b--){
            temp = nums[a];
            nums[a] = nums[b];
            nums[b] = temp;
        }

        for (int a = res, b = nums.size()-1; a < b; a++, b--){
            temp = nums[a];
            nums[a] = nums[b];
            nums[b] = temp;
        }
    }

};